system-tests/cypress/utils/selectors.utils.js

```diff
@@ -1,9 +1,40 @@
-export default {
-  LOADING_INDICATOR: '[data-testid="loader"]',
-  LOGIN_ICON: '[data-testid="LoginIcon"]',
+const placeholderBuilder = (value) => `[placeholder="${value}"]`;
+
+const testIdBuilder = (value) => `[data-testid="${value}"]`;
+
+const idBuilder = (value) => `[id="${value}"]`;
+
+const SELECTORS = {
+  LOADING_INDICATOR: testIdBuilder('loader'),
+  LOGIN_ICON: testIdBuilder('LoginIcon'),
   DEV_LOGIN_TEXT: 'Local Dev User',
-  CR_ROW: (title) => `[data-testid="${title}crRow"]`,
+  CR_ROW: (title) => testIdBuilder(`${title}crRow`),
   NEW_CHANGE_REQUEST_BUTTON: 'New Change Request',
   ALL_CHANGE_REQUESTS_TAB: 'All Change Requests',
-  CHANGE_REQUEST_TABLE: '[data-testid="Change Request Table"]'
+  CHANGE_REQUEST_TABLE: testIdBuilder('Change Request Table'),
+  PROJECT_OR_WORKPACKAGE_PLACEHOLDER: placeholderBuilder('Select a project or work package'),
+  ISSUE_BUTTON: 'ISSUE',
+  DEFINITION_CHANGE_BUTTON: 'DEFINITION_CHANGE',
+  OTHER_BUTTON: 'OTHER',
+  WHAT_DESCRIPTOR: 'What needs to be changed',
+  WHY_DESCRIPTOR: 'Why does this need to be changed',
+  WHY_TYPE_OPTION: (index) => `[name="why.${index}.type"]`,
+  WHY_EXPLAIN_TEXT_BOX: (index) => idBuilder(`why.${index}.explain-input`),
+  ADD_REASON: 'ADD REASON',
+  WHY_DELETE_OPTION: testIdBuilder('DeleteIcon'),
+  EXPLAIN_TEXT_BOX_PLACEHOLDER: placeholderBuilder('Explain *'),
+  ADD_PROPOSED_SOLUTION_BUTTON: '+ Add Solution',
+  PROPOSED_SOLUTION_DESCRIPTION_INPUT: idBuilder('description-input'),
+  PROPOSED_SOLUTION_SCOPE_INPUT: idBuilder('scopeImpact-input'),
+  PROPOSED_SOLUTION_BUDGET_INPUT: idBuilder('budgetImpact-input'),
+  PROPOSED_SOLUTION_TIMELINE_INPUT: idBuilder('timelineImpact-input'),
+  ADD_BUTTON: new RegExp('^Add$', 'g'),
+  SUBMIT_BUTTON: new RegExp('^Submit$', 'g'),
+  DIALOG: '[role="dialog"]',
+  ACTIONS_BUTTON: 'Actions',
+  ACTIONS_BUTTON_DELETE: 'Delete',
+  CONFIRM_DELETE_TEXT_INPUT: idBuilder('identifier-input'),
+  CHANGE_REQUEST_TITLE: new RegExp('Change Request #\\d+$')
```

Comment:
Do you think we should create a different list of selectors for each type of test, that might be a little cleaner as this list would start getting rlly big

Reply:
yeah probably a good idea.